source/state_representation/src/robot/Jacobian.cpp

```diff
@@ -80,6 +80,43 @@ Jacobian Jacobian::Random(const std::string& robot_name,
   return random;
 }
 
+bool Jacobian::is_compatible(const State& state) const {
+  bool compatible = false;
+  switch (state.get_type()) {
+    case StateType::JACOBIANMATRIX:
+      // compatibility is assured through the vector of joint names
+      compatible = (this->get_name() == state.get_name())
+          && (this->cols_ == dynamic_cast<const Jacobian&>(state).get_joint_names().size());
```

Comment:
What is the difference between static_cast and dynamic_cast?

Reply:
The behaviour is pretty much identical for value casting (e.g. `TypeA a = static_cast<TypeB>(b)`). In both cases, the compiler can check that the conversion is valid, and will fail the build if not.

But, if you are casting pointers or references, then the compiler cannot always check the content of the object in advance, and there may be sneaky cases where a conversion is not possible. 

Static casting does no run time checks, and therefore is a little bit faster - but if the object you are referencing does not have all the memory in all the right places (a simple case being a void pointer `void *` that's trying to be interpreted as an object pointer `Type *`), then you will have some kind of unhandled memory corruption / stack fault somewhere down the line.

Dynamic casting does an additional run-time check to verify the content of the reference or pointer is actually valid, and throws a `bad_cast` exception if not. It doesn't fix the problem, but it handles it in an explicit and catchable way.